src/acl/DestinationIp.cc

```diff
@@ -76,35 +78,27 @@ ACLDestinationIP::match(ACLChecklist *cl)
     } else if (!checklist->request->flags.destinationIpLookedUp) {
         /* No entry in cache, lookup not attempted */
         debugs(28, 3, "can't yet compare '" << name << "' ACL for " << checklist->request->url.host());
-        if (checklist->goAsync(DestinationIPLookup::Instance()))
+        if (checklist->goAsync(StartLookup, *this))
             return -1;
         // else fall through to mismatch, hiding the lookup failure (XXX)
     }
 
     return 0;
 }
 
-DestinationIPLookup DestinationIPLookup::instance_;
-
-DestinationIPLookup *
-DestinationIPLookup::Instance()
-{
-    return &instance_;
-}
-
 void
-DestinationIPLookup::checkForAsync(ACLChecklist *cl)const
+ACLDestinationIP::StartLookup(ACLChecklist &cl, const ACL &)
 {
-    ACLFilledChecklist *checklist = Filled(cl);
+    const auto checklist = Filled(&cl);
```

Comment:
If _possible_, let's change the first AsyncStarter() parameter type to an ACLFilledChecklist reference and call Filled() inside the ACLChecklist::goAsync() method. Or will that cause linking problems? Since all(?) our AsyncStarter functions call Filled(), it would be nice to do it for them in the caller...

Reply:
I moved Filled() as you suggested - it caused to linking problems.